service/src/logger.rs

```diff
@@ -54,35 +56,86 @@ impl Logger {
     }
 }
 
-impl log::Log for Logger {
-    fn enabled(&self, metadata: &Metadata) -> bool {
-        for (target, level) in self.filter.iter() {
-            if metadata.target().starts_with(target) && &metadata.level() <= level {
-                return true;
+impl<S> Layer<S> for Logger
+where
+    S: Subscriber,
+{
+    fn enabled(
+        &self,
+        metadata: &tracing::Metadata<'_>,
+        _ctx: tracing_subscriber::layer::Context<'_, S>,
+    ) -> bool {
+        metadata.level() <= &Level::INFO
+    }
+
+    fn on_event(
+        &self,
+        event: &tracing::Event<'_>,
+        _ctx: tracing_subscriber::layer::Context<'_, S>,
+    ) {
+        let datetime = Utc::now();
+        let metadata = event.metadata();
+
+        let item = {
+            let mut visitor = JsonVisitor::default();
+            event.record(&mut visitor);
+
+            // drop log metadata as it is included in the other LogItem fields (target, file, line...)
+            let fields: serde_json::Map<String, serde_json::Value> = visitor.0
+                .into_iter()
+                .filter(|(field, _)| !field.starts_with("log."))
+                .collect();
```

Comment:
Hmm, I thought about that, but did it this way for separation of concerns and to avoid duplication of this filter on every `record_x()` method in the visitor. Is this a performance hit?

Reply:
Yip, a little of a performance hit. I was thinking of adding a helper method on the visitor to do the filtering and then calling the helper in each `record_x()`

Reply:
Great, will update :+1: